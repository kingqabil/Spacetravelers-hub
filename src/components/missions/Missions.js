import { useEffect } from 'react';
import axios from 'axios';
import { useDispatch, useSelector } from 'react-redux';

import MissionsTable from './MissionsTable';
import { ADD_MISSION, createAction } from '../../redux/missions/missions';

const missionsAPI = 'https://api.spacexdata.com/v3/missions';

const populateReduxStore = (missions, dispatch) => {
  missions.forEach((mission) => {
    dispatch(createAction(ADD_MISSION, mission));
  });
};

const moveAPIDataToMissionsArray = (data, dispatch) => {
  const missionsArray = [];

  data.forEach((item) => {
    missionsArray.push({
      id: item.mission_id,
      name: item.mission_name,
      description: item.description,
      reserved: false,
    });
  });

  populateReduxStore(missionsArray, dispatch);
};

const Missions = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions);

  useEffect(() => {
    if (missions.length === 0) {
      axios
        .get(missionsAPI)
        .then((missions) => moveAPIDataToMissionsArray(missions.data, dispatch));
    }
  }, []);

  return <MissionsTable />;
};

export default Missions;
