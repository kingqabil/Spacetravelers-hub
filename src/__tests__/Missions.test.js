import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Missions from '../components/missions/Missions';

describe('Missions Component test', () => {
  it('Renders Mission Page', () => {
    const app = renderer.create(
      <Provider store={store}>
        <Missions />

      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});
