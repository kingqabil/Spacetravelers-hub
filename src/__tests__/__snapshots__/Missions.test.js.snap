// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Component test Renders Mission Page 1`] = `
<div
  className="table-wrapper"
>
  <table
    className="table table-striped table-bordered"
  >
    <thead>
      <tr>
        <th
          className="mission"
        >
          Mission
        </th>
        <th
          className="description"
        >
          Description
        </th>
        <th
          className="status"
        >
          Status
        </th>
        <th
          className="reserve"
        />
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`Rockets Component test Renders Rocket Page 1`] = `
<div
  className="rockets-list"
/>
`;

exports[`renders correctly 1`] = `
<div
  className="table-wrapper"
>
  <table
    className="table table-striped table-bordered"
  >
    <thead>
      <tr>
        <th
          className="mission"
        >
          Mission
        </th>
        <th
          className="description"
        >
          Description
        </th>
        <th
          className="status"
        >
          Status
        </th>
        <th
          className="reserve"
        />
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
